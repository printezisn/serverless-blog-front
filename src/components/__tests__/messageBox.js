import React from "react"
import renderer from "react-test-renderer"

import MessageBox from "../messageBox"

describe("MessageBox", () => {
    it("renders correctly when it's for errors and doesn't have any messages", () => {
        const box = renderer
            .create(<MessageBox isForErrors={true} messages={[]} />)
            .toJSON()
        expect(box).toMatchSnapshot()
    })

    it("renders correctly when it's for errors and has 1 message", () => {
        const box = renderer
            .create(<MessageBox isForErrors={true} messages={["msg1"]} />)
            .toJSON()
        expect(box).toMatchSnapshot()
    })

    it("renders correctly when it's for errors and has more than 1 messages", () => {
        const box = renderer
            .create(
                <MessageBox isForErrors={true} messages={["msg1", "msg2"]} />
            )
            .toJSON()
        expect(box).toMatchSnapshot()
    })

    it("renders correctly when it's not for errors and doesn't have any messages", () => {
        const box = renderer
            .create(<MessageBox isForErrors={false} messages={[]} />)
            .toJSON()
        expect(box).toMatchSnapshot()
    })

    it("renders correctly when it's not for errors and has 1 message", () => {
        const box = renderer
            .create(<MessageBox isForErrors={false} messages={["msg1"]} />)
            .toJSON()
        expect(box).toMatchSnapshot()
    })

    it("renders correctly when it's not for errors and has more than 1 messages", () => {
        const box = renderer
            .create(
                <MessageBox isForErrors={false} messages={["msg1", "msg2"]} />
            )
            .toJSON()
        expect(box).toMatchSnapshot()
    })
})
