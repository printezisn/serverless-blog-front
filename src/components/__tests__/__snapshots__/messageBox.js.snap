// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox renders correctly when it's for errors and doesn't have any messages 1`] = `""`;

exports[`MessageBox renders correctly when it's for errors and has 1 message 1`] = `
<div
  className="notification is-danger"
>
  msg1
</div>
`;

exports[`MessageBox renders correctly when it's for errors and has more than 1 messages 1`] = `
<div
  className="notification is-danger"
>
  <ul>
    <li>
      msg1
    </li>
    <li>
      msg2
    </li>
  </ul>
</div>
`;

exports[`MessageBox renders correctly when it's not for errors and doesn't have any messages 1`] = `""`;

exports[`MessageBox renders correctly when it's not for errors and has 1 message 1`] = `
<div
  className="notification is-success"
>
  msg1
</div>
`;

exports[`MessageBox renders correctly when it's not for errors and has more than 1 messages 1`] = `
<div
  className="notification is-success"
>
  <ul>
    <li>
      msg1
    </li>
    <li>
      msg2
    </li>
  </ul>
</div>
`;
